src/main/java/com/aws/greengrass/clientdevices/auth/api/Result.java

```diff
@@ -19,6 +21,12 @@ private Result(Status status, T value) {
         this.status = status;
     }
 
+    private Result(Status status, T value, Exception e) {
```

Comment:
This is done so we can throw errors from useCases and the type signature remains the same... Currently needed to allow us to move code embedded in classes to use cases without having to change the old code. We just make the old code call a use case.

Reply:
See here https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/163/files#diff-4cbc61e8fc9bd73ada6b4ddb43f7e84858fbc178aed0134d410876e0fd382362R53